### ISSUE TITLE
gstoraster calls gs with "-dPARANOIDSAFER" and this causes cups to state "Filter failed"

### ISSUE THREAD

**Issue by pezcurrel** [OpenPrinting/cups-filters#59]
Hi,
I'm running an up-to-date Arch Linux box with...

kernel 4.18.5
cups 2.2.8
libcups 2.2.8
cups-filters 1.21.1
ghostscript 9.23

My HP LaserJet Pro P1102 works ok with hplip 3.18.6, every job gets completed, but cups holds every job with a "Filter failed" message since gs, called by gstoraster, throws an error. Here are the relevant lines from /var/log/cups/error_log

```
D [02/Sep/2018:14:24:02 +0200] [Job 18] 3 filters for job:
D [02/Sep/2018:14:24:02 +0200] [Job 18] pdftopdf (application/pdf to application/vnd.cups-pdf, cost 66)
D [02/Sep/2018:14:24:02 +0200] [Job 18] gstoraster (application/vnd.cups-pdf to application/vnd.cups-raster, cost 99)
D [02/Sep/2018:14:24:02 +0200] [Job 18] hpcups (application/vnd.cups-raster to printer/Stampante_che_funziona, cost 0)

D [02/Sep/2018:14:24:03 +0200] [Job 18] Ghostscript command line: gs -dQUIET -dPARANOIDSAFER -dNOPAUSE -dBATCH -dNOINTERPOLATE -dNOMEDIAATTRS -dShowAcroForm -sstdout=%stderr -sOutputFile=%stdout -sDEVICE=cups -sMediaType=Plain -sOutputType=0 -r600x600 -dMediaPosition=7 -dDEVICEWIDTHPOINTS=612 -dDEVICEHEIGHTPOINTS=792 -dcupsBitsPerColor=1 -dcupsColorOrder=0 -dcupsColorSpace=3 -dcupsRowCount=1 -dcupsRowStep=2 -dcupsInteger0=2 -scupsPageSizeName=Letter -I/usr/share/cups/fonts -c \'<</.HWMargins[12.000000 12.000000 12.000000 12.000000] /Margins[0 0]>>setpagedevice\' -f -_

D [02/Sep/2018:14:24:03 +0200] [Job 18] Error: /invalidfileaccess in --run--
D [02/Sep/2018:14:24:03 +0200] [Job 18] Operand stack:
D [02/Sep/2018:14:24:03 +0200] [Job 18] true   (/var/spool/cups/tmp/gs_FzErtn)
D [02/Sep/2018:14:24:03 +0200] [Job 18] Execution stack:
D [02/Sep/2018:14:24:03 +0200] [Job 18] %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push   2015   2   3   %oparray_pop   2014   2   3   %oparray_pop   1998   2   3   %oparray_pop
D [02/Sep/2018:14:24:03 +0200] [Job 18] Dictionary stack:
D [02/Sep/2018:14:24:03 +0200] [Job 18] --dict:1001/1684(ro)(G)--   --dict:1/20(G)--   --dict:78/200(L)--
D [02/Sep/2018:14:24:03 +0200] [Job 18] Current allocation mode is local
D [02/Sep/2018:14:24:03 +0200] [Job 18] Last OS error: No such file or directory
D [02/Sep/2018:14:24:03 +0200] [Job 18] GPL Ghostscript 9.23: Unrecoverable error, exit code 1
D [02/Sep/2018:14:24:03 +0200] [Job 18] Rendering completed
```

By testing from terminal the gs command above, I've seen that only by omitting the "-dPARANOIDSAFER" option or replacing it with "-dNOSAFER" the command doesn't throw the error (with "-dSAFER" it does, and ghostscript devs plan to set "-dSAFER" as a default sometimes in the future).
Just to test, I then edited the gstoraster binary with an hex editor replacing "-dPARANOIDSAFER" with "-dNOSAFER" and now gs exits with no error and cups doesn't hold the jobs complaining about "Filter failed".

**Comment by tillkamppeter** [comment]
Perhaps there is a relationship with this bug in Ghostscript:
https://bugs.ghostscript.com/show_bug.cgi?id=699713

**Comment by tillkamppeter** [comment]
No, there is no relationship of this bug to mentioned Ghostscript bug, but in the GIT repo of Ghostscript there is a fix for this bug:
http://git.ghostscript.com/?p=ghostpdl.git;a=commitdiff;h=0704d18b1031
I am not sure whether it is included in the now released Ghostscript 9.24.

**Comment by tillkamppeter** [comment]
I have confirmation now that the fix is in GS 9.24. Please update and check.
Closing as this is a Ghostscript bug.

**Comment by pezcurrel** [comment]
Checked with GS 9.25-2 (Arch), it's fixed :)
Many thanks
